### PR TITLE
Sysdig

### PR DESCRIPTION
**Pull request by BryanGK**
This PR adds in the custom resource for sysdig monitoring.

### CODE REVIEW

**Comment by wenzowski** [comment]
The rest of our k8s yaml is being managed via helm; would it make sense for us to start a tools chart? I'd love to keep that configuration as code approach and use github actions to ensure the config in this repo becomes the deployed config automatically as part of our release cycle

File: .github/workflows/publish_deploy_image.yaml [hunk 1]
```diff
@@ -106,7 +106,7 @@ jobs:
 
   deploy-to-openshift-test:
     if: github.event.ref == 'refs/heads/main'
-    needs: [build-and-push-image]
+    needs: [tests, build-and-push-image]
```
Comment (wenzowski) [inline .github/workflows/publish_deploy_image.yaml:109]: While this additional pipeline improvement isn't about Sysdig, I agree the test env should have the same stage requirements that dev env has

File: helm/tools/templates/deployer/deployerRole.yaml [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+      - sysdig-teams
+    verbs:
+      - delete
+      - deletecollection
+      - get
+      - list
+      - patch
+      - create
+      - update
+      - watch
```
Comment (wenzowski) [inline helm/tools/templates/deployer/deployerRole.yaml:187]: This is probably the only role our service account needs to have right now, right? Granting our role permissions we're not going to use is a [Least Privilege](https://cheatsheetseries.owasp.org/cheatsheets/Authorization_Cheat_Sheet.html#enforce-least-privileges) violation. Can we try deleting the other roles and making sure our service account can still deploy?

**Review by wenzowski — CHANGES_REQUESTED** [review]
Really really close, just looking to tighten up the role grant in the service account

File: .github/workflows/publish_deploy_image.yaml (hunk 1 above)
Comment (BryanGK) [reply .github/workflows/publish_deploy_image.yaml:109]: should I revert this and open a new ticket for it?

Comment (wenzowski) [reply .github/workflows/publish_deploy_image.yaml:109]: No I think this is a great change. Usually would be a separate PR, yes, but leave it in :)

**Review by wenzowski — APPROVED** [review]
Great work!

**Pull request merged**